crates/client/src/lib.rs

```diff
@@ -405,12 +399,21 @@ impl<R: RegistryStorage, C: ContentStorage> Client<R, C> {
                 })?;
 
             for record in response.operator {
-                let record: PublishedProtoEnvelope<operator::OperatorRecord> = record.try_into()?;
-                operator
-                    .state
-                    .validate(&record.envelope)
-                    .map_err(|inner| ClientError::OperatorValidationFailed { inner })?;
-                operator.head_registry_index = Some(record.registry_index);
+                let proto_envelope: PublishedProtoEnvelope<operator::OperatorRecord> =
+                    record.envelope.try_into()?;
+                if operator.head_registry_index.is_none()
+                    || operator.head_registry_index.is_some_and(|registry_index| {
+                        proto_envelope.registry_index > registry_index
+                    })
+                {
```

Comment:
Alternative wording:

```suggestion
                if proto_envelope.registry_index > operator.head_registry_index.unwrap_or(0) {
```

Reply:
Thanks for the suggestion, still getting better at Rust. I'll fix. Doesn't quite work I think given it is an unsigned integer and `0` is a valid (and likely registry index for the Operator log). Seeing if it can be cleaner.

Reply:
So I can simplifiy it, to `operator.head_registry_index.is_none() || proto_envelope.registry_index > operator.head_registry_index.unwrap()`. Since the `.is_none() || ` will ensure the `.unwrap()` won't panic.